src/anndata/_core/aligned_mapping.py

```diff
@@ -419,15 +424,17 @@ def __init__(
 PairwiseArraysBase._actual_class = PairwiseArrays
 
 
-class AlignedMappingProperty(property):
-    def __init__(self, name, cls, axis):
+class AlignedMappingProperty(property, Generic[T]):
+    def __init__(
+        self, name: str, cls: T, axis: Literal[0, 1] | tuple[Literal[0], Literal[1]]
```

Comment:
Isn't `cls` of `Type[T]`?

Reply:
currently T is inferred to be `type[...]` for each descriptor, e.g.

![grafik](https://github.com/scverse/anndata/assets/291575/d696428d-bd44-4766-860b-c3724613cbc3)

so either would be correct.